Apps/W1/HybridGP/app/src/Migration/Customers/GPCustomerMigrator.codeunit.al

```diff
@@ -375,4 +375,80 @@ codeunit 4018 "GP Customer Migrator"
         HelperFunctions.UpdateFieldWithValue(RecordVariant, MigrationGPCustTrans.FieldNo(GLDocNo), DocumentNo);
     end;
 
+    procedure MigrateCustomerClasses()
+    var
+        GPCompanyAdditionalSettings: Record "GP Company Additional Settings";
+        GPRM00101: Record "GP RM00101";
+        GPRM00201: Record "GP RM00201";
+        CustomerPostingGroup: Record "Customer Posting Group";
+        Customer: Record Customer;
+        HelperFunctions: Codeunit "Helper Functions";
+        ClassId: Text[20];
+        AccountNumber: Code[20];
+        MigrateCustomerClasses: Boolean;
+    begin
+        if not GPRM00101.FindSet() then
+            exit;
+
+        MigrateCustomerClasses := GPCompanyAdditionalSettings.GetMigrateCustomerClasses();
+        if not MigrateCustomerClasses then
+            exit;
+
+        repeat
+            Clear(GPRM00201);
+            Clear(CustomerPostingGroup);
+
+            ClassId := GPRM00101.CUSTCLAS.Trim();
+            if ClassId <> '' then
+                if Customer.Get(GPRM00101.CUSTNMBR) then begin
+                    if not CustomerPostingGroup.Get(ClassId) then
+                        if GPRM00201.Get(ClassId) then begin
```

Comment:
Suggestion - is it better to move the customer posting groups first, regardless, then to assign them as we create customers? We could save on the SQL queries.

Reply:
Good suggestion, I made the change.